govc/flags/client.go

```diff
@@ -625,3 +627,29 @@ func (flag *ClientFlag) Environ(extra bool) []string {
 
 	return env
 }
+
+// WithCancel calls the given function, returning when complete or canceled via SIGINT.
+func (flag *ClientFlag) WithCancel(ctx context.Context, f func(context.Context) error) error {
+	wctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	done := make(chan bool)
+	var werr error
+
+	go func() {
+		defer close(done)
+		werr = f(wctx)
+	}()
+
+	sig := make(chan os.Signal)
```

Comment:
That reference doesn't say why though.  Go's docs are a bit better: https://golang.org/pkg/os/signal/#Notify
Your comment here also made me realize the `sig` channel needs to be created earlier, before the go routine calling `f` is started.  Changed so that happens first.